update.rb

```diff
@@ -97,158 +97,162 @@
+          conflict_merge_request_id = omr.iid
+          break
+        end
+      end
     end
-  end
 
-  # Enable GitLab "merge when pipeline succeeds" feature.
-  # Merge requests created and successfully tested will be merge automatically.
-  if ENV["DEPENDABOT_GITLAB_AUTO_MERGE"]
-    begin
-      gitlab_client.accept_merge_request(
-        source.repo,
-        merge_request_id,
-        merge_when_pipeline_succeeds: true,
-        should_remove_source_branch: true
+    merge_request_id = nil
+    if conflict_merge_request_commit_id && conflict_merge_request_id
+      ########################################
+      # Update merge request with conflict   #
+      ########################################
+      pr_updater = Dependabot::PullRequestUpdater.new(
+        source: source,
+        base_commit: commit,
+        old_commit: conflict_merge_request_commit_id,
+        files: updated_files,
+        credentials: credentials,
+        pull_request_number: conflict_merge_request_id,
       )
-    rescue Exception => e
-      print "\nError when trying to merge the merge request\n#{e.message}"
+      pr_updater.update
+      merge_request_id = conflict_merge_request_id
+      print " merge request ##{conflict_merge_request_id} updated"
     else
-      print " / set to be accepted"
+      ########################################
+      # Create a pull request for the update #
+      ########################################
+      pr_creator = Dependabot::PullRequestCreator.new(
+        source: source,
+        base_commit: commit,
+        dependencies: updated_deps,
+        files: updated_files,
+        credentials: credentials,
+        label_language: true,
+        assignees: assignees
+      )
+      pull_request = pr_creator.create
+      merge_request_id = pull_request.iid if pull_request
+      print " submitted"
+    end
+
+    opened_merge_requests += 1
+    next unless merge_request_id
+
+    # Auto approve Gitlab merge request with the same user.
+    if ENV["DEPENDABOT_GITLAB_APPROVE_MERGE"]
+      begin
+        gitlab_client.approve_merge_request(source.repo, merge_request_id)
+      rescue Exception => e
+        print "\nError when trying to approve the merge request\n#{e.message}"
+      else
+        print " / approved"
+      end
+    end
+
+    # Enable GitLab "merge when pipeline succeeds" feature.
+    # Merge requests created and successfully tested will be merge automatically.
+    if ENV["DEPENDABOT_GITLAB_AUTO_MERGE"]
+      begin
+        gitlab_client.accept_merge_request(
+          source.repo,
+          merge_request_id,
+          merge_when_pipeline_succeeds: true,
+          should_remove_source_branch: true
+        )
+      rescue Exception => e
+        print "\nError when trying to merge the merge request\n#{e.message}"
+      else
+        print " / set to be accepted"
+      end
     end
+  rescue Exception => e
+    puts "error updating #{dep.name}"
```

Comment:
We should probably still add an option to fail the whole build if anything fails.